woocommerce/woocommerce-ios
# Add support for force-refreshing the server-side analytics cache

Closes: #7055 


### Description


On the server side, our API team [has a plan](https://github.com/woocommerce/woocommerce/issues/33315) to update how the analytics cache is invalidated so that it's invalidated less often to potentially improve site performance. This means that after the changes on the cache, the data in the wc-analytics API response could be outdated more often. This is fine for API requests triggered like when switching a time range tab, but there are times when we still want to always fetch the latest data like when the user pulls to refresh or on the first view of a stats time range tab. In order to distinguish whether a "force-refresh" is required, a new parameter `force_cache_refresh` was added in WC 6.7.0 https://github.com/woocommerce/woocommerce/pull/33325 and we need to implement this parameter before the API team can make any changes to the cache.

This PR thus added the new parameter `force_cache_refresh` to two API requests: order stats and leaderboards. The parameter doesn't make a difference right now, but will be necessary when the cache changes are implemented in a future WC version.

### Testing instructions


I don't think there's a way to verify the new param works, asked about this in peaMlT-X-p2#comment-47 but haven't got a response.

Please just make sure the stats are loaded correctly before for all 4 time range tabs in the My store tab, for sites with WC version 6.7.0+.


---
